tests/unit/s2n_client_psk_extension_test.c

```diff
@@ -399,226 +391,141 @@ int main(int argc, char **argv)
-        {
-            struct s2n_connection *conn;
-            EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_SERVER));
-
-            struct s2n_psk_identity psk_identities[2] = { 0 };
-            psk_identities[0].data = test_bytes_data;
-            psk_identities[0].length = sizeof(test_bytes_data);
-            psk_identities[1].data = test_bytes_data_2;
-            psk_identities[1].length = sizeof(test_bytes_data_2);
-
-            struct s2n_psk *no_match_psk = NULL;
-            EXPECT_OK(s2n_array_pushback(&conn->psk_params.psk_list, (void **) &no_match_psk));
-            EXPECT_OK(s2n_psk_init(no_match_psk, S2N_PSK_TYPE_EXTERNAL));
-            EXPECT_SUCCESS(s2n_psk_set_identity(no_match_psk, test_zero_bytes, sizeof(test_zero_bytes)));
-
-            struct s2n_psk *match_psk = NULL;
-            EXPECT_OK(s2n_array_pushback(&conn->psk_params.psk_list, (void **) &match_psk));
-            EXPECT_OK(s2n_psk_init(match_psk, S2N_PSK_TYPE_EXTERNAL));
-            EXPECT_SUCCESS(s2n_psk_set_identity(match_psk, test_bytes_data_2, sizeof(test_bytes_data_2)));
-
-            size_t psk_identities_length = sizeof(psk_identities)/sizeof(struct s2n_psk_identity);
-
-            EXPECT_OK(s2n_select_psk_identity(conn, psk_identities, psk_identities_length));
-            EXPECT_EQUAL(conn->psk_params.chosen_psk_wire_index, 1);
-            EXPECT_EQUAL(conn->psk_params.chosen_psk, match_psk);
-
-            EXPECT_SUCCESS(s2n_connection_free(conn));
-        }
+        struct s2n_blob identity_1 = { 0 };
+        EXPECT_SUCCESS(s2n_blob_init(&identity_1, test_identity, sizeof(test_identity)));
+        struct s2n_blob identity_2 = { 0 };
+        EXPECT_SUCCESS(s2n_blob_init(&identity_2, test_identity_2, sizeof(test_identity_2)));
+        struct s2n_blob identity_3 = { 0 };
+        EXPECT_SUCCESS(s2n_blob_init(&identity_3, test_identity_3, sizeof(test_identity_3)));
+
+        struct s2n_blob *all_psks_list[] = { &identity_1, &identity_2, &identity_3 };
+        struct s2n_blob *reverse_order_list[] = { &identity_3, &identity_2, &identity_1 };
+        struct s2n_blob *list_without_psk1[] = { &identity_2, &identity_3 };
+        struct s2n_blob *multiple_psk1[] = { &identity_1, &identity_1, &identity_1 };
+        struct s2n_blob *multiple_psk2[] = { &identity_2, &identity_2, &identity_2 };
+        struct s2n_blob *only_psk1[] = { &identity_1 };
+        struct s2n_blob *only_psk2[] = { &identity_2 };
+        struct s2n_blob *only_psk3[] = { &identity_3 };
+
+        struct {
+            bool match;
+            struct s2n_blob **wire_identities;
+            size_t wire_identities_len;
+            struct s2n_blob **local_identities;
+            size_t local_identities_len;
+            size_t wire_match_index;
+            size_t local_match_index;
+        } test_cases[] = {
+#define WIRE_IDENTITIES(list) .wire_identities = list, .wire_identities_len = s2n_array_len(list)
```

Comment:
To match the new name change to `offered_psks` we could change `WIRE_IDENTITIES` to  `OFFERED_IDENTITIES` to keep it consistent.

Reply:
The names aren't intended to match the structures, they're intended to describe the environment we're testing in. WIRE vs LOCAL is much clearer than OFFERED_PSKS vs PSK_PARAMS when describing which PSKs we choose. Do you think WIRE vs LOCAL isn't clear?